src/main/java/sorald/Repair.java

```diff
@@ -196,22 +202,46 @@ private void writeModel(CtModel model, Path outputDir) {
 
         boolean isIntermediateOutputDir =
                 outputDir.toString().contains(intermediateSpoonedPath.toString());
-        if (config.getFileOutputStrategy() == FileOutputStrategy.ALL || isIntermediateOutputDir) {
+        FileOutputStrategy outputStrategy = config.getFileOutputStrategy();
+        if (outputStrategy == FileOutputStrategy.ALL || isIntermediateOutputDir) {
             processingManager.process(model.getUnnamedModule().getFactory().Class().getAll());
         } else {
-            assert (config.getFileOutputStrategy() == FileOutputStrategy.CHANGED_ONLY
-                    && !isIntermediateOutputDir);
+            assert outputStrategy == FileOutputStrategy.CHANGED_ONLY
+                    || outputStrategy == FileOutputStrategy.IN_PLACE;
 
             for (Map.Entry<String, CtType> patchedFile :
                     UniqueTypesCollector.getInstance().getTopLevelTypes4Output().entrySet()) {
-                javaOutputProcessor.process(patchedFile.getValue());
+
+                if (outputStrategy == FileOutputStrategy.CHANGED_ONLY) {
+                    javaOutputProcessor.process(patchedFile.getValue());
+                } else {
+                    assert outputStrategy == FileOutputStrategy.IN_PLACE;
+                    CtType<?> type = patchedFile.getValue();
+                    String output =
+                            type.getFactory()
+                                    .getEnvironment()
+                                    .createPrettyPrinter()
+                                    .printTypes(type);
+                    writeString(Paths.get(patchedFile.getKey()), output);
+                }
+
                 if (config.getGitRepoPath() != null) {
                     createPatches(patchedFile.getKey(), javaOutputProcessor);
                 }
             }
         }
     }
 
+    private static void writeString(Path filepath, String output) {
+        try {
+            Files.writeString(filepath, output);
+        } catch (IOException e) {
+            // most convert to a runtime exception as this is used in writeModel, which in turn
+            // is used in a stream (that can't have checked exceptions)
+            throw new RuntimeException(e);
```

Comment:
Coverage drop is due to this case never being hit. We must convert the IOException to a RuntimeException (or any derivative of that) as this method is indirectly used in a stream. Streams cannot have checked exceptions thrown in them as they are lazy, and therefore can be evaluated somewhere other than where they are defined.